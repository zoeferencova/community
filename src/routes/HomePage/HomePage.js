import React, { Component } from "react";
import { Link, withRouter } from "react-router-dom";
import { PropTypes } from 'prop-types';
import { ButtonLight } from "../../components/Utils/Utils";
import MyPost from "../../components/MyPost/MyPost";
import Post from "../../components/Post/Post";
import GoogleMap from "../../components/GoogleMap/GoogleMap";
import MapErrorBoundary from "../../components/MapErrorBoundary/MapErrorBoundary";
import CommUnityContext from "../../contexts/context";
import styles from "./HomePage.module.css";

class HomePage extends Component {
    static contextType = CommUnityContext;

    state = {
        showPosts: false,
        showLocation: false,
    }

    makeUserPosts() {
        return this.context.user_posts.map(post => <MyPost key={post.id} id={post.id} post_type={post.post_type} date_created={post.date_created} />)
    }

    makeAllPosts() {
        return this.context.neighborhood_posts.map(post => <Post key={post.id} id={post.id} post_type={post.post_type} first_name={post.first_name} categories={post.categories} description={post.description} urgency={post.urgency} date_created={post.date_created} location={post.location} distance_from_user={post.distance_from_user} />)
    }

    goToNewPost = type => {
        this.props.history.push(`/new-post/${type}`)
    }
    
    render() {
        return (   
            <main className={styles.main}>
                {!this.props.loading && <>
                <header className={styles.header}>
                    <div className={styles.userInfo}>
                        <h2>Welcome, {this.context.user.first_name}!</h2>
                        <div className={styles.newPostButtons}>
                            <ButtonLight className={styles.requestButton} onClick={() => this.goToNewPost("request")}><i className={`fas fa-hand-paper ${styles.hand}`}></i> Request</ButtonLight>
                            <ButtonLight className={styles.newPostButton} onClick={() => this.goToNewPost("offer")}><i className={`fas fa-heart ${styles.heart}`}></i> Offer</ButtonLight>
                        </div>
                        <div className={styles.yourPosts}>
                            <div className={styles.infoTitle} onClick={() => this.state.showPosts ? this.setState({ showPosts: false }) : this.setState({ showPosts: true })}>
                                <h3>Your Posts</h3>
                                <p className={`${styles.plus} ${!this.state.showPosts && styles.unrotate} ${this.state.showPosts && styles.rotate}`}>+</p>
                            </div>
                            <div className={`${styles.infoBody} ${!this.state.showPosts && styles.hide}`}>
                                {this.context.user_posts.length ? this.makeUserPosts() : "You do not have any posts"}
                            </div>
                        </div>
                    </div>
                    <div className={`${styles.mapSection}`}>
                        <div className={styles.infoTitle} onClick={() => this.state.showLocation ? this.setState({ showLocation: false }) : this.setState({ showLocation: true })}>
                            <h3 className={styles.mapTitle}>Your Location</h3>
                            <p className={`${styles.plus} ${!this.state.showLocation && styles.unrotate} ${this.state.showLocation && styles.rotate}`}>+</p>
                        </div>
                        <div className={`${styles.infoBody} ${styles.infoBodyMap} ${!this.state.showLocation && styles.hide}`}>
                            <MapErrorBoundary>
                                <GoogleMap className={styles.map} radius={parseInt(this.context.user.radius)} userLocation={this.context.user.location} displayMarker={true} />
                            </MapErrorBoundary>
                            <p className={styles.mapSubtitle}>Results are being shown for this area. <Link to="/location" className={styles.mapSubLink}>Change location?</Link></p>
                        </div>
                    </div>
                </header>
                <section className={styles.feed}>
                    {this.context.neighborhood_posts.length ? <div className={styles.feedHeader}>
                        <h3>There are {this.context.neighborhood_posts.length} posts in your area</h3>
                    </div> : ""}
                    {this.makeAllPosts()}
                    {!this.context.neighborhood_posts.length && 
                        <div className={styles.noPosts}>
                            <img src={require("../../images/neighborhood.png")} alt="neighborhood" className={styles.noPostImage}></img>
                            <p>There are no posts in your area.<br></br><span className={styles.noPostTip}>Try <Link to="/location">increasing your radius</Link> to see more posts.</span></p>
                        </div>
                    }
                </section>
                </>}
            </main>
        )
    }
}

export default withRouter(HomePage);

HomePage.propTypes = {
    loading: PropTypes.bool
}