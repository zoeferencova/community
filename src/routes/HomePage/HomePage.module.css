.main {
    position: absolute;
    top: 65px;
    width: 100%;
    display: flex;
    flex-direction: column;
}

.header {
    padding: 50px 15px;
    display: flex;
    padding-bottom: 80px;
    justify-content: space-evenly;
    flex-wrap: wrap;
    align-items: center;
}

.header h2 {
    font-size: 30px;
}

.heart, .hand {
    margin-right: 7px;
}

.heart {
    color: #bf5e6b;
}

.hand {
    color: #e6c78e;
}

.newPostButtons {
    margin-bottom: 20px;
    width: 100%;
    display: flex;
    justify-content: space-between;
}

.newPostButtons button {
    width: 50%;
}

.requestButton {
    margin-right: 10px;
}

.yourPosts {
    margin-bottom: 30px;
}

.yourPosts h3, .mapTitle {
    text-align: left;
    border-bottom: 1px solid black;
    padding-bottom: 10px;
    font-size: 14px;
    text-transform: uppercase;
    letter-spacing: 2.5px;
}

.mapSection {
    width: 80%;
    height: 250px;
}

.userInfo {
    width: 80%;
}

.mapSection p {
    margin: 10px 0;
    padding-bottom: 15px;
}

.feedHeader h2 {
    font-size: 14px;
    text-transform: uppercase;
    letter-spacing: 2.5px;
}

.feedHeader {
    display: flex;
    flex-direction: column-reverse;
    width: 100%;
    text-align: left;
}

.feed {
    display: flex;
    width: 90%;
    flex-direction: column;
    align-items: center;
    padding: 50px 0px;
    margin: 0 auto;
}

@media only screen and (min-width: 700px) {
    .mapSection {
        width: 90%;
        height: 200px;
        padding-top: 0;
        margin-top: 0;
    }

    .feed {
        margin-top: 20px;
    }

    .yourPosts {
        margin: 60px 0 40px 0;
    }

    .header {
        text-align: left;
        flex-direction: column;
        margin-top: 20px;
        min-width: 260px;
        width: 35%;
        padding: 20px;
        margin-left: 5%;
        align-items: flex-start;
        justify-content: flex-start;
    }

    .newPostButtons {
        margin: 0;
        width: 60%;
    }

    .userInfo {
        width: 90%;
    }

    .main {
        flex-direction: row;
    }

    .feedHeader {
        flex-direction: row;
        width: 70%;
        justify-content: space-between;
        align-items: flex-end;
        margin-bottom: 20px;
    }

    .feedHeader h2 {
        margin-bottom: -5px;
    }
}