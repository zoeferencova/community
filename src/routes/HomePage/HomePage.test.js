import React from "react";
import { shallow, mount, render } from "enzyme";
import HomePage from "./HomePage";
import { BrowserRouter } from "react-router-dom";
import CommUnityContext from "../../contexts/context";

import mockContext from "../../contexts/mock-context";

it("expect to render HomePage component", () => {
    const wrapper = render(<CommUnityContext.Provider value={mockContext}><BrowserRouter><HomePage /></BrowserRouter></CommUnityContext.Provider>)
    
    expect(wrapper).toMatchSnapshot();
})

