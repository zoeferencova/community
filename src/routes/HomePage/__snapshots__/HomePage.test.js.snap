// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders HomePage component given mock context data 1`] = `
<main
  class="main"
>
  <header
    class="header"
  >
    <div
      class="userInfo"
    >
      <h2>
        Welcome, Zoe!
      </h2>
      <div
        class="newPostButtons"
      >
        <button
          class="button requestButton"
        >
          <i
            class="fas fa-hand-paper hand"
          />
           Request
        </button>
        <button
          class="button newPostButton"
        >
          <i
            class="fas fa-heart heart"
          />
           Offer
        </button>
      </div>
      <div
        class="yourPosts"
      >
        <div
          class="infoTitle"
        >
          <h3>
            Your Posts
          </h3>
          <p
            class="plus unrotate false"
          >
            +
          </p>
        </div>
        <div
          class="infoBody hide"
        >
          <a
            class="link"
            href="/my-post/1"
          >
            <p
              class="userPost"
            >
              <i
                class="fas fa-hand-paper hand"
              />
               Request created 7 days ago
            </p>
          </a>
          <a
            class="link"
            href="/my-post/1"
          >
            <p
              class="userPost"
            >
              <i
                class="fas fa-heart heart"
              />
               Offer created 6 days ago
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      class="mapSection"
    >
      <div
        class="infoTitle"
      >
        <h3
          class="mapTitle"
        >
          Your Location
        </h3>
        <p
          class="plus unrotate false"
        >
          +
        </p>
      </div>
      <div
        class="infoBody infoBodyMap hide"
      >
        <div
          id="google-map"
          style="height:100%;width:100%"
        />
        <p
          class="mapSubtitle"
        >
          Results are being shown for this area. 
          <a
            class="mapSubLink"
            href="/location"
          >
            Change location?
          </a>
        </p>
      </div>
    </div>
  </header>
  <section
    class="feed"
  >
    <div
      class="feedHeader"
    >
      <h3>
        There are 2 posts in your area
      </h3>
    </div>
    <div
      class="post"
    >
      <div
        class="details"
      >
        <div>
          <div
            class="postHeader"
          >
            <div
              class="nameSection"
            >
              <div
                class="circle indigo "
                first_name="James"
                user_name="Zoe"
              >
                J
              </div>
              <h4>
                James offered to help
              </h4>
            </div>
            <div>
              <span>
                <i
                  class="fas fa-heart heart"
                />
              </span>
              <span
                class="distance"
              >
                0.6 mi
              </span>
            </div>
          </div>
          <ul
            class="helpItems"
          >
            <li
              class="task supplies"
            >
              Picking up supplies
            </li>
          </ul>
          <p
            class="description"
          />
        </div>
      </div>
      <button
        class="button buttonLink"
        type="button"
      >
        Accept offer
      </button>
    </div>
    <div
      class="post"
    >
      <div
        class="details"
      >
        <div>
          <div
            class="postHeader"
          >
            <div
              class="nameSection"
            >
              <div
                class="circle red "
                first_name="Anna"
                user_name="Zoe"
              >
                A
              </div>
              <h4>
                Anna requested help
              </h4>
            </div>
            <div>
              <span>
                <i
                  class="fas fa-hand-paper hand"
                />
              </span>
              <span
                class="distance"
              >
                1.4 mi
              </span>
            </div>
          </div>
          <ul
            class="helpItems"
          >
            <li
              class="task dog"
            >
              Dog walking
            </li>
            <li
              class="task errands"
            >
              Running errands
            </li>
          </ul>
          <p
            class="description"
          />
        </div>
      </div>
      <button
        class="button buttonLink"
        type="button"
      >
        Offer to help
      </button>
    </div>
  </section>
</main>
`;
